plugins/modules/files/json_patch.py

```diff
@@ -0,0 +1,578 @@
+        """Perform a 'test' operation.
+
+        This operation supports an additional feature not outlined in
+        RFC 6901 (https://tools.ietf.org/html/rfc6901): The ability to
+        reference every member of an array by using an asterisk (*).
+
+        In such a case, each member of the array at that point in the
+        path will be tested sequentially for the given value, and if
+        a matching value is found, the method will return immediately
+        with a True value.
+
+        Example:
+            {"op": "test", "path": "/array/*/member/property", "value": 2}
+
+        If the object to be tested looked like...
+        {
+            "array": [
+                {"member": {"property": 1}},
+                {"member": {"property": 2}}
+            ]
+        }
+        ... the result would be True, because an object exists within
+        "array" that has the matching path and value.
+        """
+        elements = path.lstrip('/').split('/')
+        next_obj = obj
+        for idx, elem in enumerate(elements):
+            if elem == "*":  # wildcard
+                if not isinstance(next_obj, list):
+                    return obj, None, False
+                for sub_obj in next_obj:
+                    dummy, chg, found = self.test('/'.join(elements[(idx + 1):]), value, sub_obj)
+                    if found:
+                        return obj, None, found
+                return obj, None, False
+            try:
+                next_obj = next_obj[elem]
+            except KeyError:
+                return obj, None, False
+            except TypeError:  # it's a list
+                if not elem.isdigit():
+                    return obj, None, False
+                try:
+                    next_obj = next_obj[int(elem)]
+                except IndexError:
+                    return obj, None, False
+        return obj, None, next_obj == value
+
+
+def main():
+    # Parsing argument file
+    module = basic.AnsibleModule(
+        argument_spec=dict(
+            src=dict(required=True, type='str'),
+            dest=dict(required=False, type='str'),
+            operations=dict(
+                required=True,
+                type='list',
+                elements=dict(
+                    op=dict(required=True, type='str', choices=['add', 'remove', 'replace', 'move', 'copy', 'test']),
+                    path=dict(required=True, type='str'),
+                    value=dict(required=False, type='raw'),
+                )
+            ),
+            backup=dict(required=False, default=False, type='bool'),
+            unsafe_writes=dict(required=False, default=False, type='bool'),
+            pretty=dict(required=False, default=False, type='bool'),
+            create=dict(required=False, default=False, type='bool'),
+            create_type=dict(required=False, default='object', type='str', choices=['object', 'array']),
+            supports_check_mode=True
```

Comment:
This must be an argument to `AnsibleModule`, and not be a part of the argument spec dictionary.